src/Compilers/Core/Portable/Compilation/SymbolInfo.cs

```diff
@@ -98,9 +98,14 @@ public override bool Equals(object? obj)
 
         public bool Equals(SymbolInfo other)
         {
-            return object.Equals(this.Symbol, other.Symbol)
-                && ((_candidateSymbols.IsDefault && other._candidateSymbols.IsDefault) || _candidateSymbols.SequenceEqual(other._candidateSymbols))
-                && this.CandidateReason == other.CandidateReason;
+            if (!object.Equals(this.Symbol, other.Symbol) ||
+                _candidateSymbols.IsDefault != _candidateSymbols.IsDefault ||
```

Comment:
Is this correct? Isn't 
```
_candidateSymbols.IsDefault != _candidateSymbols.IsDefault
```
 always `false` as in `A != A` being always `false` or am I missing something here? I would expect to see something like:
 ```
_candidateSymbols.IsDefault != other._candidateSymbols.IsDefault
```
.

Reply:
Yep. That is a silly mistake. Fixing.